app/Location/Dig/HauntedGrove.php

```diff
@@ -1,175 +1,270 @@
+				$credit = "fears nothing";
+				break;
+
+			case Item::get('BigKey'):
+			case Item::get('Key'):
+				$credit = "picks locks";
+				break;
+
+			case Item::get('Compass'):
+				$credit = "finds boss";
+				break;
+
+			case Item::get('Map'):
+				$credit = "navigates";
+				break;
+
+			case Item::get('BossHeartContainer'):
+			case Item::get('HeartContainer'):
+			case Item::get('HeartContainerNoAnimation'):
+			case Item::get('PieceOfHeart'):
+				$credit = "feels love";
+				break;
+
+			case Item::get('Bomb'):
+			case Item::get('TenBombs'):
+			case Item::get('ThreeBombs'):
+				$credit = "explodes";
+				break;
+
+			case Item::get('Mushroom'):
+				$credit = "sells drugs";
+				break;
+
+			case Item::get('Bottle'):
+				$credit = "has terrarium";
+				break;
+
+			case Item::get('BottleWithRedPotion'):
+				$credit = "has red goo";
+				break;
+
+			case Item::get('BottleWithGreenPotion'):
+				$credit = "has green goo";
+				break;
+
+			case Item::get('BottleWithBluePotion'):
+				$credit = "has blue goo";
+				break;
+
+			case Item::get('BottleWithGoldBee'):
+				$credit = "has beetor";
+				break;
+
+			case Item::get('BottleWithBee'):
+				$credit = "has mad bee";
+				break;
+
+			case Item::get('BottleWithFairy'):
+				$credit = "has friend";
+				break;
+
+			case Item::get('Heart'):
+				$credit = "has affection";
+				break;
+
+			case Item::get('Arrow'):
+			case Item::get('TenArrows'):
+				$credit = "sews";
+				break;
+
+			case Item::get('SmallMagic'):
+				$credit = "summons";
+				break;
+
+			case Item::get('FiftyRupees'):
+			case Item::get('FiveRupees'):
+			case Item::get('OneRupee'):
+			case Item::get('TwentyRupees'):
+			case Item::get('TwentyRupees2'):
+				$credit = "has lunch";
+				break;
+
+			case Item::get('OneHundredRupees'):
+				$credit = "goes drinking";
+				break;
+
+			case Item::get('ThreeHundredRupees'):
+				$credit = "is rich";
+				break;
+
+			case Item::get('PegasusBoots'):
+				$credit = "runs";
+				break;
+
+			case Item::get('BombUpgrade5'):
+			case Item::get('BombUpgrade10'):
+			case Item::get('BombUpgrade50'):
+				$credit = "explodes more";
+				break;
+
+			case Item::get('ArrowUpgrade5'):
+			case Item::get('ArrowUpgrade10'):
+			case Item::get('ArrowUpgrade70'):
+				$credit = "sews more";
+				break;
+
+			case Item::get('SilverArrowUpgrade'):
+				$credit = "sharpens";
+				break;
+
+			case Item::get('HalfMagic'):
+			case Item::get('QuarterMagic'):
+				$credit = "saves magic";
+				break;
+
+			case Item::get('Rupoor'):
+				$credit = "steals rupees";
+				break;
+
+			case Item::get('RedClock'):
+				$credit = "travels time";
+				break;
+
+			case Item::get('BlueClock'):
+				$credit = "time travels";
+				break;
+
+			case Item::get('GreenClock'):
+				$credit = "adjusts time";
+				break;
+
+			case Item::get('Nothing'):
+				$credit = "does nothing";
+				break;
+
+		}
+
+		$credit = "boy " . $credit . " again";
+
+		if($this->item == Item::get('Hammer')) {

```

Comment:
strangely this feels more complex to me than a simple switch that is returning static text